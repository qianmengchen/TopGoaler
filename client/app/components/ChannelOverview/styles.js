import { StyleSheet } from 'react-native';

const board = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    marginHorizontal: 10
  },
  channels: {
    flexDirection: 'row',
    borderWidth: 2,
    borderColor: 'black'
  }
});

const avatar = StyleSheet.create({
  summary: {
    marginLeft: 40
  }
});

const text = StyleSheet.create({
  title: {
    fontSize: 30,
    fontWeight: 'bold',
    paddingLeft: 10
  },
  subtitle: {
    fontSize: 20,
    paddingLeft: 10
  },
  desc: {
    fontSize: 20,
    paddingLeft: 10
  }
});

export { avatar, board, text };
