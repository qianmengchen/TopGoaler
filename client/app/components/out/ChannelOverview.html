<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ChannelOverview</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ChannelOverview</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ChannelOverview<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ChannelOverview"><span class="type-signature"></span>new ChannelOverview<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>channel_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last">A unique number identifying a channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>subscribe</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function allowing the user to subscribe to a specific channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>channel</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last">An object describing a specific channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>title</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The name of the channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>task</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            

            

            <td class="description last">An object describing a specific task.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>user_channel</code></td>
            

            <td class="type">
            
                
<span class="param-type">array</span>


            
            </td>

            

            

            <td class="description last">An array of channels that the user is subscribed to.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>is_subscribed</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">A boolean value indicating whether the user is subscribed or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>goToChannel</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function that allows the user to navigate to the clicked channel.</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="ChannelOverview.js.html">ChannelOverview.js</a>, <a href="ChannelOverview.js.html#line21">line 21</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="ChannelOverview.html">ChannelOverview</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sun Dec 02 2018 21:01:58 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>