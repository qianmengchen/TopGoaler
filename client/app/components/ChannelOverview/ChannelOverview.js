import React, { Component } from 'react';
import {
  View,
  TouchableHighlight,
  TouchableWithoutFeedback
} from 'react-native';
import { Avatar, Text, Icon } from 'react-native-elements';
import { board, contents, desc } from './styles';

class ChannelOverview extends Component {
  _submit = () => {
    let { goToChannel, channel } = this.props;
    goToChannel(channel.id);
  };

  _subscribe = () => {
    let { subscribe, title } = this.props;
    subscribe(title);
  };

  //utils function
  _count_users_reducer = (acc, val) => {
    if (val.channel_id === this.props.channel.id) {
      return acc + 1;
    } else {
      return acc;
    }
  };

  _number_of_users = () => {
    return this.props.user_channel.reduce(this._count_users_reducer, 0);
  };

  _count_tasks_reducer = (acc, val) => {
    if (val.channel_id === this.props.channel.id) {
      return acc + 1;
    } else {
      return acc;
    }
  };

  _number_of_tasks = () => {
    return this.props.task.reduce(this._count_tasks_reducer, 0);
  };

  render() {
    let button;
    if (this.props.is_subscribed) {
      button = (
        <TouchableWithoutFeedback>
          <Icon style={desc.icon} name="check" size={30} color="black" />
        </TouchableWithoutFeedback>
      );
    } else {
      button = (
        <TouchableWithoutFeedback onPress={this._subscribe.bind(this)}>
          <Icon style={desc.icon} name="add" size={30} color="black" />
        </TouchableWithoutFeedback>
      );
    }

    return (
      <TouchableHighlight
        underlayColor="#aaa"
        onPress={this._submit.bind(this)}
      >
        <View style={board.container}>
          <View style={board.contents}>
            <Text style={contents.title}>{this.props.channel.title}</Text>
            <Text style={contents.subtitle}>{this.props.channel.subtitle}</Text>
            <View style={contents.desc}>
              <Text
                style={desc.numbers}
              >{`${this._number_of_users()} users, ${this._number_of_tasks()} tasks`}</Text>
              {button}
            </View>
          </View>
          <View style={board.avatar}>
            <Avatar
              size={80}
              source={{
                uri: this.props.channel.image_url
              }}
            />
          </View>
        </View>
      </TouchableHighlight>
    );
  }
}

export default ChannelOverview;
