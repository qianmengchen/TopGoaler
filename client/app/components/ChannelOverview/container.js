import { connect } from 'react-redux';
import ChannelOverview from './ChannelOverview';
import { createChannelAsUser, subscribeChannel } from '../../actions';

const mapStateToProps = state => ({
  user_channel: state.user_channel
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ChannelOverview);
