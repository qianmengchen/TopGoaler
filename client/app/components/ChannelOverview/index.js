import ChannelOverview from './container';

export { ChannelOverview };
