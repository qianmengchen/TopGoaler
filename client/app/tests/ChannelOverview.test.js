import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import { Avatar, Text } from 'react-native-elements';
require('isomorphic-fetch');

import ChannelOverview from '../components/ChannelOverview/ChannelOverview';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const _goToChannel = jest.fn();
const _subscribe = jest.fn();

const ch = {
  id: 42,
  title: 'bxzhu_channel',
  creator: 1,
  subtitle: 'bxzhu',
  image_url: 'http://shortlink.in/themes/v3/styles/img/url-link.png'
};

const user_channel = [];
const task = [];

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing Channel Overview Page', () => {
  const wrapper = shallow(
    <ChannelOverview
      key={0}
      channel={ch}
      goToChannel={() => _goToChannel()}
      subscribe={() => _subscribe()}
      user_channel={user_channel}
      task={task}
    />,
    {
      context: { store: mockStore(initialTaskState) }
    }
  );
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should have 1 Avatar for channel logo', () => {
    expect(render.find(Avatar)).toHaveLength(1);
  });

  it('should have 3 Text: title, subtitle, description', () => {
    expect(render.find(Text)).toHaveLength(3);
  });

  it('should have 1 TouchableWithoutFeedback for channel logo', () => {
    expect(render.find('TouchableWithoutFeedback')).toHaveLength(1);
  });

  it('should invoke correct methods when pressing submit button', () => {
    const submitBtn = render.find('TouchableHighlight');
    submitBtn.simulate('press');
    expect(_goToChannel).toHaveBeenCalledTimes(1);
  });
});
