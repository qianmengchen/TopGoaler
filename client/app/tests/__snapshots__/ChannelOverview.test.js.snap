// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Channel Overview Page renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChannelOverview
    channel={
      Object {
        "creator": 1,
        "id": 42,
        "image_url": "http://shortlink.in/themes/v3/styles/img/url-link.png",
        "subtitle": "bxzhu",
        "title": "bxzhu_channel",
      }
    }
    goToChannel={[Function]}
    subscribe={[Function]}
    task={Array []}
    user_channel={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.85,
      "children": <Component
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 6,
            "borderWidth": 0,
            "elevation": 1,
            "flex": 1,
            "flexDirection": "row",
            "height": 120,
            "marginBottom": 10,
            "shadowColor": "#888",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 1,
          }
        }
      >
        <Component
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 0.75,
              "flexDirection": "column",
              "justifyContent": "center",
              "paddingLeft": 20,
            }
          }
        >
          <Themed.Text
            style={
              Object {
                "alignItems": "center",
                "flex": 0.3,
                "fontFamily": "Avenir-Heavy",
                "fontSize": 20,
                "paddingBottom": 15,
                "paddingTop": 20,
              }
            }
          >
            bxzhu_channel
          </Themed.Text>
          <Themed.Text
            style={
              Object {
                "alignItems": "center",
                "flex": 0.35,
                "fontFamily": "Avenir-Light",
                "fontSize": 16,
              }
            }
          >
            bxzhu
          </Themed.Text>
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 0.35,
                "flexDirection": "row",
                "paddingBottom": 10,
              }
            }
          >
            <Themed.Text
              style={
                Object {
                  "alignItems": "center",
                  "color": "#555",
                  "flex": 0.7,
                  "fontFamily": "Helvetica-Light",
                  "justifyContent": "center",
                }
              }
            >
              0 users, 0 tasks
            </Themed.Text>
            <TouchableWithoutFeedback
              onPress={[Function]}
            >
              <Themed.Icon
                color="black"
                name="add"
                size={30}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.3,
                    "justifyContent": "center",
                    "paddingBottom": 10,
                  }
                }
              />
            </TouchableWithoutFeedback>
          </Component>
        </Component>
        <Component
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.25,
              "justifyContent": "center",
              "paddingRight": 20,
            }
          }
        >
          <Themed.Avatar
            size={80}
            source={
              Object {
                "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
              }
            }
          />
        </Component>
      </Component>,
      "delayPressOut": 100,
      "onPress": [Function],
      "underlayColor": "#aaa",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.75,
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingLeft": 20,
              }
            }
          >
            <Themed.Text
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0.3,
                  "fontFamily": "Avenir-Heavy",
                  "fontSize": 20,
                  "paddingBottom": 15,
                  "paddingTop": 20,
                }
              }
            >
              bxzhu_channel
            </Themed.Text>
            <Themed.Text
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0.35,
                  "fontFamily": "Avenir-Light",
                  "fontSize": 16,
                }
              }
            >
              bxzhu
            </Themed.Text>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0.35,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                }
              }
            >
              <Themed.Text
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#555",
                    "flex": 0.7,
                    "fontFamily": "Helvetica-Light",
                    "justifyContent": "center",
                  }
                }
              >
                0 users, 0 tasks
              </Themed.Text>
              <TouchableWithoutFeedback
                onPress={[Function]}
              >
                <Themed.Icon
                  color="black"
                  name="add"
                  size={30}
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 0.3,
                      "justifyContent": "center",
                      "paddingBottom": 10,
                    }
                  }
                />
              </TouchableWithoutFeedback>
            </Component>
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.25,
                "justifyContent": "center",
                "paddingRight": 20,
              }
            }
          >
            <Themed.Avatar
              size={80}
              source={
                Object {
                  "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
                }
              }
            />
          </Component>,
        ],
        "style": Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 6,
          "borderWidth": 0,
          "elevation": 1,
          "flex": 1,
          "flexDirection": "row",
          "height": 120,
          "marginBottom": 10,
          "shadowColor": "#888",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Themed.Text
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.3,
                    "fontFamily": "Avenir-Heavy",
                    "fontSize": 20,
                    "paddingBottom": 15,
                    "paddingTop": 20,
                  }
                }
              >
                bxzhu_channel
              </Themed.Text>,
              <Themed.Text
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.35,
                    "fontFamily": "Avenir-Light",
                    "fontSize": 16,
                  }
                }
              >
                bxzhu
              </Themed.Text>,
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.35,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                  }
                }
              >
                <Themed.Text
                  style={
                    Object {
                      "alignItems": "center",
                      "color": "#555",
                      "flex": 0.7,
                      "fontFamily": "Helvetica-Light",
                      "justifyContent": "center",
                    }
                  }
                >
                  0 users, 0 tasks
                </Themed.Text>
                <TouchableWithoutFeedback
                  onPress={[Function]}
                >
                  <Themed.Icon
                    color="black"
                    name="add"
                    size={30}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 0.3,
                        "justifyContent": "center",
                        "paddingBottom": 10,
                      }
                    }
                  />
                </TouchableWithoutFeedback>
              </Component>,
            ],
            "style": Object {
              "alignItems": "flex-start",
              "flex": 0.75,
              "flexDirection": "column",
              "justifyContent": "center",
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "bxzhu_channel",
                "style": Object {
                  "alignItems": "center",
                  "flex": 0.3,
                  "fontFamily": "Avenir-Heavy",
                  "fontSize": 20,
                  "paddingBottom": 15,
                  "paddingTop": 20,
                },
              },
              "ref": null,
              "rendered": "bxzhu_channel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "bxzhu",
                "style": Object {
                  "alignItems": "center",
                  "flex": 0.35,
                  "fontFamily": "Avenir-Light",
                  "fontSize": 16,
                },
              },
              "ref": null,
              "rendered": "bxzhu",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Themed.Text
                    style={
                      Object {
                        "alignItems": "center",
                        "color": "#555",
                        "flex": 0.7,
                        "fontFamily": "Helvetica-Light",
                        "justifyContent": "center",
                      }
                    }
                  >
                    0 users, 0 tasks
                  </Themed.Text>,
                  <TouchableWithoutFeedback
                    onPress={[Function]}
                  >
                    <Themed.Icon
                      color="black"
                      name="add"
                      size={30}
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 0.3,
                          "justifyContent": "center",
                          "paddingBottom": 10,
                        }
                      }
                    />
                  </TouchableWithoutFeedback>,
                ],
                "style": Object {
                  "alignItems": "center",
                  "flex": 0.35,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "0 users, 0 tasks",
                    "style": Object {
                      "alignItems": "center",
                      "color": "#555",
                      "flex": 0.7,
                      "fontFamily": "Helvetica-Light",
                      "justifyContent": "center",
                    },
                  },
                  "ref": null,
                  "rendered": "0 users, 0 tasks",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Themed.Icon
                      color="black"
                      name="add"
                      size={30}
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 0.3,
                          "justifyContent": "center",
                          "paddingBottom": 10,
                        }
                      }
                    />,
                    "onPress": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "color": "black",
                      "name": "add",
                      "size": 30,
                      "style": Object {
                        "alignItems": "center",
                        "flex": 0.3,
                        "justifyContent": "center",
                        "paddingBottom": 10,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Themed.Avatar
              size={80}
              source={
                Object {
                  "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
                }
              }
            />,
            "style": Object {
              "alignItems": "flex-end",
              "flex": 0.25,
              "justifyContent": "center",
              "paddingRight": 20,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "size": 80,
              "source": Object {
                "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.85,
        "children": <Component
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "black",
              "borderRadius": 6,
              "borderWidth": 0,
              "elevation": 1,
              "flex": 1,
              "flexDirection": "row",
              "height": 120,
              "marginBottom": 10,
              "shadowColor": "#888",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 1,
            }
          }
        >
          <Component
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.75,
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingLeft": 20,
              }
            }
          >
            <Themed.Text
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0.3,
                  "fontFamily": "Avenir-Heavy",
                  "fontSize": 20,
                  "paddingBottom": 15,
                  "paddingTop": 20,
                }
              }
            >
              bxzhu_channel
            </Themed.Text>
            <Themed.Text
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0.35,
                  "fontFamily": "Avenir-Light",
                  "fontSize": 16,
                }
              }
            >
              bxzhu
            </Themed.Text>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0.35,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                }
              }
            >
              <Themed.Text
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#555",
                    "flex": 0.7,
                    "fontFamily": "Helvetica-Light",
                    "justifyContent": "center",
                  }
                }
              >
                0 users, 0 tasks
              </Themed.Text>
              <TouchableWithoutFeedback
                onPress={[Function]}
              >
                <Themed.Icon
                  color="black"
                  name="add"
                  size={30}
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 0.3,
                      "justifyContent": "center",
                      "paddingBottom": 10,
                    }
                  }
                />
              </TouchableWithoutFeedback>
            </Component>
          </Component>
          <Component
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.25,
                "justifyContent": "center",
                "paddingRight": 20,
              }
            }
          >
            <Themed.Avatar
              size={80}
              source={
                Object {
                  "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
                }
              }
            />
          </Component>
        </Component>,
        "delayPressOut": 100,
        "onPress": [Function],
        "underlayColor": "#aaa",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.75,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingLeft": 20,
                }
              }
            >
              <Themed.Text
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.3,
                    "fontFamily": "Avenir-Heavy",
                    "fontSize": 20,
                    "paddingBottom": 15,
                    "paddingTop": 20,
                  }
                }
              >
                bxzhu_channel
              </Themed.Text>
              <Themed.Text
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.35,
                    "fontFamily": "Avenir-Light",
                    "fontSize": 16,
                  }
                }
              >
                bxzhu
              </Themed.Text>
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0.35,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                  }
                }
              >
                <Themed.Text
                  style={
                    Object {
                      "alignItems": "center",
                      "color": "#555",
                      "flex": 0.7,
                      "fontFamily": "Helvetica-Light",
                      "justifyContent": "center",
                    }
                  }
                >
                  0 users, 0 tasks
                </Themed.Text>
                <TouchableWithoutFeedback
                  onPress={[Function]}
                >
                  <Themed.Icon
                    color="black"
                    name="add"
                    size={30}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 0.3,
                        "justifyContent": "center",
                        "paddingBottom": 10,
                      }
                    }
                  />
                </TouchableWithoutFeedback>
              </Component>
            </Component>,
            <Component
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 0.25,
                  "justifyContent": "center",
                  "paddingRight": 20,
                }
              }
            >
              <Themed.Avatar
                size={80}
                source={
                  Object {
                    "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
                  }
                }
              />
            </Component>,
          ],
          "style": Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 6,
            "borderWidth": 0,
            "elevation": 1,
            "flex": 1,
            "flexDirection": "row",
            "height": 120,
            "marginBottom": 10,
            "shadowColor": "#888",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Themed.Text
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 0.3,
                      "fontFamily": "Avenir-Heavy",
                      "fontSize": 20,
                      "paddingBottom": 15,
                      "paddingTop": 20,
                    }
                  }
                >
                  bxzhu_channel
                </Themed.Text>,
                <Themed.Text
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 0.35,
                      "fontFamily": "Avenir-Light",
                      "fontSize": 16,
                    }
                  }
                >
                  bxzhu
                </Themed.Text>,
                <Component
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 0.35,
                      "flexDirection": "row",
                      "paddingBottom": 10,
                    }
                  }
                >
                  <Themed.Text
                    style={
                      Object {
                        "alignItems": "center",
                        "color": "#555",
                        "flex": 0.7,
                        "fontFamily": "Helvetica-Light",
                        "justifyContent": "center",
                      }
                    }
                  >
                    0 users, 0 tasks
                  </Themed.Text>
                  <TouchableWithoutFeedback
                    onPress={[Function]}
                  >
                    <Themed.Icon
                      color="black"
                      name="add"
                      size={30}
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 0.3,
                          "justifyContent": "center",
                          "paddingBottom": 10,
                        }
                      }
                    />
                  </TouchableWithoutFeedback>
                </Component>,
              ],
              "style": Object {
                "alignItems": "flex-start",
                "flex": 0.75,
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingLeft": 20,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "bxzhu_channel",
                  "style": Object {
                    "alignItems": "center",
                    "flex": 0.3,
                    "fontFamily": "Avenir-Heavy",
                    "fontSize": 20,
                    "paddingBottom": 15,
                    "paddingTop": 20,
                  },
                },
                "ref": null,
                "rendered": "bxzhu_channel",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "bxzhu",
                  "style": Object {
                    "alignItems": "center",
                    "flex": 0.35,
                    "fontFamily": "Avenir-Light",
                    "fontSize": 16,
                  },
                },
                "ref": null,
                "rendered": "bxzhu",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Themed.Text
                      style={
                        Object {
                          "alignItems": "center",
                          "color": "#555",
                          "flex": 0.7,
                          "fontFamily": "Helvetica-Light",
                          "justifyContent": "center",
                        }
                      }
                    >
                      0 users, 0 tasks
                    </Themed.Text>,
                    <TouchableWithoutFeedback
                      onPress={[Function]}
                    >
                      <Themed.Icon
                        color="black"
                        name="add"
                        size={30}
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 0.3,
                            "justifyContent": "center",
                            "paddingBottom": 10,
                          }
                        }
                      />
                    </TouchableWithoutFeedback>,
                  ],
                  "style": Object {
                    "alignItems": "center",
                    "flex": 0.35,
                    "flexDirection": "row",
                    "paddingBottom": 10,
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "0 users, 0 tasks",
                      "style": Object {
                        "alignItems": "center",
                        "color": "#555",
                        "flex": 0.7,
                        "fontFamily": "Helvetica-Light",
                        "justifyContent": "center",
                      },
                    },
                    "ref": null,
                    "rendered": "0 users, 0 tasks",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Themed.Icon
                        color="black"
                        name="add"
                        size={30}
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 0.3,
                            "justifyContent": "center",
                            "paddingBottom": 10,
                          }
                        }
                      />,
                      "onPress": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "color": "black",
                        "name": "add",
                        "size": 30,
                        "style": Object {
                          "alignItems": "center",
                          "flex": 0.3,
                          "justifyContent": "center",
                          "paddingBottom": 10,
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Themed.Avatar
                size={80}
                source={
                  Object {
                    "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
                  }
                }
              />,
              "style": Object {
                "alignItems": "flex-end",
                "flex": 0.25,
                "justifyContent": "center",
                "paddingRight": 20,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "size": 80,
                "source": Object {
                  "uri": "http://shortlink.in/themes/v3/styles/img/url-link.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
